sympy/utilities/misc.py

```diff
@@ -355,7 +355,8 @@ def translate(s, a, b=None, c=None):
 
     mr = {}
     if a is None:
-        assert c is None
+        if c is not None:
+            raise TypeError('Called translate with 4 arguments and a second argument of None, expected 3 arguments if second argument is None')
```

Comment:
SymPy is develop with good programming practice. Look at `PEP 8`: https://www.python.org/dev/peps/pep-0008/#id19
In Python, line should have below 80 characters